controllers/usersignup/usersignup_controller.go

```diff
@@ -655,7 +657,14 @@ func (r *Reconciler) ensureNewMurIfApproved(reqLogger logr.Logger, config toolch
 	}
 
 	// Provision the MasterUserRecord
-	return r.provisionMasterUserRecord(reqLogger, config, userSignup, targetCluster, userTier)
+	if err := r.provisionMasterUserRecord(reqLogger, config, userSignup, targetCluster, userTier); err != nil {
+		return err
+	}
+	// track activation in Segment
+	if r.SegmentClient != nil {
```

Comment:
I wonder if we can add some tests for that. I guess segment provides some mock libs for testing.

Reply:
good point, let me check that!

Reply:
I couldn't find any mock to test the Segment client, but I refactored the code in addressed in e701dec to optionally replace it in the unit tests with a mock instance that we can use to assert that the message was queued